### ISSUE TITLE
Enhancement : create user with a login instead of an email adress

### ISSUE THREAD

**Issue by GLLM** [Sonerezh/sonerezh#55]
I much prefer having users defined with login+password instead of email+password.
The email adress should be optional.

Thanks :)

**Comment by lGuillaume124** [comment]
Hi,
We have made a choice. We will use email address in future releases (for example if you want to retrieve a forgotten password, or send other notifications).

**Comment by GLLM** [comment]
I note your point.

I'm sad/disappointed to see that many enhancement/features get dismissed without discussion.
But you're the project leaders, so I'll do with it.

**Comment by lGuillaume124** [comment]
What would you like to discuss?

**Comment by nodiscc** [comment]
I agree with @GLLM though this is not a massive problem. Email should be optional, people may want to a username which is common to their other self-hosted apllications. Users may want not to disclose their e-mail address to the server admin. Users may not trust their mail provider with password reset tokens. Anyway I personally don't have an MTA to send mail from my server. Usernames should be freeform (alphanumeric and some symbols)

**Comment by p1rox** [comment]
+1 @nodiscc

**Comment by AC10** [comment]
:+1: 
I like using username, rather than email. Especially with something like this, where I'm planning on running it for personal use only

**Comment by GLLM** [comment]
@nodiscc said it all :)

@lGuillaume124 I didn't mean to be rude btw ...

**Comment by lGuillaume124** [comment]
@GLLM No no that's my fault. Using email to authenticate users was an evident thing for both of us. But you have good arguments, and we are sensible to the end-user privacy stakes. 

On the other hand, your requests requires some deep changes and we will not be able to propose them before the 1.1.0 or 1.2.0 release.
